src/main/scala/io/scalechain/blockchain/BlockDataInputStream.scala

```diff
@@ -1,18 +1,42 @@
 package io.scalechain.blockchain
 
 import java.io.{InputStream, DataInputStream}
+import collection.mutable.HashMap
 
 import io.scalechain.blockchain.util.Utils
 
+class FieldStatsCalculator {
+  val fieldMap = new HashMap[String, Long]()
```

Comment:
I actually have no knowledge of this project so what I'm about say would be complete non-sense. If so, please forgive me and ignore it.

Isn't it better to have immutable `Map` and make `fieldMap` `var`? Since `fieldMap` is publicly visible and mutable, more than one thread can mutate it which may cause [race condition](https://en.wikipedia.org/wiki/Race_condition).

So I think it should probably be either immutable or not publicly available.

Is the mutable Map used because of performance?

Well, as I said, depending on how it's used (e.g. it's only used in single thread and never be shared with any other threads), my questions and suggestions would just sound ignorant. Sorry, I'm doing it out of curiosity. 😅